packages/main/src/ColorPalette.ts

```diff
@@ -376,55 +402,62 @@ class ColorPalette extends UI5Element {
 			this._handleDefaultColorClick(e);
 		}
 
-		if (isDown(e)) {
+		if (this._isNext(e)) {
 			e.stopPropagation();
-
-			this.focusColorElement(this.colorPaletteNavigationElements[1], this._itemNavigation);
+			this._focusFirstDisplayedColor();
+		} else if (isLeft(e)) {
+			e.stopPropagation();
+			this._focusFirstAvailable(
+				() => this._focusLastRecentColor(),
+				() => this._focusMoreColors(),
+				() => this._focusLastDisplayedColor(),
+			);
 		} else if (isUp(e)) {
 			e.stopPropagation();
-			const lastElementInNavigation = this.colorPaletteNavigationElements[this.colorPaletteNavigationElements.length - 1];
-
-			if (this.hasRecentColors) {
-				this.focusColorElement(lastElementInNavigation, this._itemNavigationRecentColors);
-			} else if (this.showMoreColors) {
-				lastElementInNavigation.focus();
-			} else {
-				const colorPaletteFocusIndex = (this.displayedColors.length % this.rowSize) * this.rowSize;
-
-				this.focusColorElement(this.displayedColors[colorPaletteFocusIndex], this._itemNavigation);
-			}
+			this._focusFirstAvailable(
+				() => this._focusLastRecentColor(),
```

Comment:
why do we need these wrappers?

Reply:
That is an early return technique. The `_focusFirstAvailable` will execute sequentially those functions, until one of them succeeds to place focus and returns `true`.